javaslang-test/src/test/java/javaslang/test/ArbitraryTest.java

```diff
@@ -185,6 +189,79 @@ public int nextInt(int bound) {
         assertThat(actual.length()).isEqualTo(10);
     }
 
+    @Test
+    public void shouldCreateArbitraryLocalDateTime(){
+        final Arbitrary<LocalDateTime> date = Arbitrary.localDateTime();
+
+        assertThat(date).isNotNull();
+    }
+
+
+    @Test(expected = NullPointerException.class)
+    public void shouldNotAcceptNullMedianLocalDateTime(){
+        final Arbitrary<LocalDateTime> dates = Arbitrary.localDateTime(null, ChronoUnit.DAYS);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldNotAcceptNullChronoUnit(){
+        final Arbitrary<LocalDateTime> dates = Arbitrary.localDateTime(LocalDateTime.now(), null);
+    }
+
+    @Test
+    public void shouldCreateArbitraryLocalDateTimeAdjustedWithGivenChronoUnit(){
+        final LocalDateTime median = LocalDateTime.of(2017, 2, 17, 3, 40);
+        final Arbitrary<LocalDateTime> arbitrary = Arbitrary.localDateTime(median, ChronoUnit.YEARS);
+
+        final LocalDateTime date = arbitrary.apply(100).apply(predictableRandom);
+
+        assertThat(date).isEqualTo("2063-04-22T01:46:10.312");
+    }
+
+    @Test
+    public void shouldCreateMedianLocalDateTimeIfSizeIsZero(){
+        final LocalDateTime median = LocalDateTime.now();
+
+        final Arbitrary<LocalDateTime> arbitrary = Arbitrary.localDateTime(median, ChronoUnit.DAYS);
+
+        final LocalDateTime date = arbitrary.apply(0).apply(RANDOM);
+
+        assertThat(date).isEqualTo(median);
+    }
+
+    @Test
+    public void shouldCreateDatesInInRangeOfSize(){
+        final LocalDateTime median = LocalDateTime.now();
+        final Arbitrary<LocalDateTime> arbitrary = Arbitrary.localDateTime(median, ChronoUnit.DAYS);
+
+        Property.def("With negative size of -100 days, dates should be in range of +/- 100 days")
+                .forAll(arbitrary)
+                .suchThat(d -> d.isAfter(median.minusDays(100)) && d.isBefore(median.plusDays(100)))
+                .check(100, 1000);
+    }
+
+    @Test
+    public void shouldIgnoreNegativeSignInRangeOfDates(){
+        final LocalDateTime median = LocalDateTime.now();
+        final Arbitrary<LocalDateTime> arbitrary = Arbitrary.localDateTime(median, ChronoUnit.DAYS);
+
+        Property.def("With size of 100 days, dates should be in range of +/- 100 days")
+                .forAll(arbitrary)
+                .suchThat(d -> d.isAfter(median.minusDays(100)) && d.isBefore(median.plusDays(100)))
+                .check(-100, 1000);
+    }
+
+    @Test
+    public void shouldGenerateTwoDifferentSuccessiveDates(){
+        final LocalDateTime end = LocalDateTime.now().minusDays(2);
```

Comment:
\[FindBugs\] INFO: DLS: Dead store to end in javaslang\.test\.ArbitraryTest\.shouldGenerateTwoDifferentSuccessiveDates\(\)